Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PostCampaignPledgeRewardsSummaryViewController.swift

```diff
@@ -0,0 +1,182 @@
+    super.viewDidLoad()
+
+    self.configureSubviews()
+    self.setupConstraints()
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  private func configureSubviews() {
+    _ = (self.rootStackView, self.view)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToEdgesInParent()
+
+    _ = (
+      [self.tableViewContainer, self.separatorView, self.pledgeTotalViewController.view],
+      self.rootStackView
+    )
+      |> ksr_addArrangedSubviewsToStackView()
+
+    _ = (self.tableView, self.tableViewContainer)
+      |> ksr_addSubviewToParent()
+
+    self.tableView.registerCellClass(PostCampaignPledgeRewardsSummaryHeaderCell.self)
+    self.tableView.registerCellClass(PostCampaignPledgeRewardsSummaryCell.self)
+  }
+
+  private func setupConstraints() {
+    let tableViewContainerHeightConstraint = self.tableViewContainer.heightAnchor
+      .constraint(equalToConstant: 0)
+    self.tableViewContainerHeightConstraint = tableViewContainerHeightConstraint
+
+    NSLayoutConstraint.activate([
+      tableViewContainerHeightConstraint,
+      self.tableView.leftAnchor.constraint(equalTo: self.tableViewContainer.leftAnchor),
+      self.tableView.rightAnchor.constraint(equalTo: self.tableViewContainer.rightAnchor),
+      self.tableView.topAnchor.constraint(equalTo: self.tableViewContainer.topAnchor),
+      self.tableViewContainer.leftAnchor.constraint(equalTo: self.rootStackView.leftAnchor),
+      self.tableViewContainer.rightAnchor.constraint(equalTo: self.rootStackView.rightAnchor),
+      self.tableViewContainer.topAnchor.constraint(equalTo: self.rootStackView.topAnchor),
+      self.separatorView.leftAnchor
+        .constraint(equalTo: self.rootStackView.leftAnchor, constant: Styles.grid(4)),
+      self.separatorView.rightAnchor
+        .constraint(equalTo: self.rootStackView.rightAnchor, constant: -Styles.grid(4)),
+      self.separatorView.heightAnchor.constraint(equalToConstant: 1),
+      self.rootStackView.widthAnchor.constraint(equalTo: self.view.widthAnchor)
+    ])
+  }
+
+  // MARK: - Bind Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = self.view
+      |> \.clipsToBounds .~ true
+      |> checkoutWhiteBackgroundStyle
+      |> \.isHidden .~ true
+
+    _ = self.rootStackView
+      |> self.rootStackViewStyle
+
+    _ = self.tableView
+      |> checkoutWhiteBackgroundStyle
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+
+    _ = self.separatorView
+      |> self.separatorViewStyle
+
+    _ = self.pledgeTotalViewController.view
+      |> \.isHidden .~ true
+  }
+
+  // MARK: - View model
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.loadRewardsIntoDataSource
+      .observeForUI()
+      .observeValues { [weak self] data in
+        guard let self else { return }
+
+        self.dataSource.load(data)
+        self.tableView.reloadData()
+        self.tableView.setNeedsLayout()
+
+        DispatchQueue.main.async {
```

Comment:
What's the story behind this dispatch here? Does this need to be `async`?

Reply:
Usually I consider an `async` in UI code to be slightly suspicious, so I recommend throwing in a comment explaining why it's necessary if it is.